internal/wasmldr/src/loader/compiled/tls.rs

```diff
@@ -0,0 +1,340 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use std::any::Any;
+use std::io::{ErrorKind, IoSlice, IoSliceMut, Read, Write};
+use std::mem::forget;
+use std::ops::{Deref, DerefMut};
+use std::sync::{Arc, RwLock};
+
+use cap_std::net::{TcpListener as CapListener, TcpStream as CapStream};
+use rustix::fd::{AsRawFd, FromRawFd};
+use rustls::{ClientConfig, ClientConnection, Connection, ServerConfig, ServerConnection};
+use wasi_common::file::{Advice, FdFlags, FileType, Filestat};
+use wasi_common::{Context, Error, ErrorExt, WasiFile};
+use wasmtime_wasi::net::{TcpListener as AnyListener, TcpStream as AnyStream};
+
+struct Forgotten<T>(Option<T>);
+
+impl<T> Deref for Forgotten<T> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        self.0.as_ref().unwrap()
+    }
+}
+
+impl<T> DerefMut for Forgotten<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        self.0.as_mut().unwrap()
+    }
+}
+
+impl<T> From<T> for Forgotten<T> {
+    fn from(value: T) -> Self {
+        Self(Some(value))
+    }
+}
+
+impl<T> Drop for Forgotten<T> {
+    fn drop(&mut self) {
+        forget(self.0.take());
```

Comment:
Is the entire purpose of this type just to leak anything it wraps? What does this achieve? A comment explaining the purpose of this type would be useful.

Reply:
Done.